src/views/project/institutions/CreateView.vue

```diff
@@ -19,21 +19,43 @@ onMounted(() => {
   }
 })
 
-async function addInstitution(institutionId, index) {
+async function addInstitution(name) {
+  // list of viable institutions from the database
+  let institution = searchList.value.find(
+    (institution) => institution.name == name
+  )
+
+  if (institution == undefined) {
+    return createInstitution(name)
+  }
+
   const success = await projectInstitutionsStore.addInstitution(
     projectId,
-    institutionId
+    institution.institution_id
   )
 
   if (success) {
-    searchList.value.splice(index, 1)
-
     await router.push({ path: `/myprojects/${projectId}/institutions` })
   } else {
     alert('Failed to Assign Institution')
   }
 }
 
+async function createInstitution(name) {
+  const url = `${
+    import.meta.env.VITE_API_URL
+  }/projects/${projectId}/institutions/createInstitution`
+  const response = await axios.post(url, { name })
```

Comment:
Let's move all the methods that modify the `projectInstitutionsStore` into the class itself.